tests/slist_ut/slist_ut.c

```diff
@@ -0,0 +1,1114 @@
+TEST_FUNCTION(slist_add_with_multiple_entries_in_the_list_succeeds)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+    static simpleItem simp2 = { 2, { NULL } };
+    static simpleItem simp3 = { 3, { NULL } };
+    static simpleItem simp4 = { 4, { NULL } };
+
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+    head.next = &(simp1.link);
+    head.next->next = &(simp2.link);
+    head.next->next->next = &(simp3.link);
+
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_add(&head, &(simp4.link));
+
+    // assert
+    ASSERT_IS_NOT_NULL(result);
+    ASSERT_ARE_EQUAL(void_ptr, &(simp4.link), result);
+    ASSERT_ARE_EQUAL(void_ptr, simp3.link.next, result);
+}
+
+/*slist_add_head*/
+
+/* Tests_SRS_SLIST_07_008: [ If `list_head` is `NULL`, `slist_add_head` shall fail and return `NULL`. ]*/
+TEST_FUNCTION(slist_add_head_with_head_NULL_fails)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+   
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_add_head(NULL, &(simp1.link));
+
+    // assert
+    ASSERT_ARE_EQUAL(void_ptr, NULL, result);
+}
+
+/* Tests_SRS_SLIST_07_009: [ If `list_entry` is `NULL`, `slist_add_head` shall fail and return `NULL`. ]*/
+TEST_FUNCTION(slist_add_head_with_add_entry_NULL_fails)
+{
+    //arrange
+    static simpleItem simp1 = { 1, { NULL } };
+    static simpleItem simp2 = { 2, { NULL } };
+
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+    slist_add(&head, &(simp1.link));
+    slist_add(&head, &(simp2.link));
+
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_add_head(&head, NULL);
+
+    // assert
+    ASSERT_ARE_EQUAL(void_ptr, NULL, result);
+}
+
+/* Tests_SRS_SLIST_07_010: [ `slist_add_head` shall insert `list_entry` at head on success and return an entry to the added entry. ]*/
+TEST_FUNCTION(slist_add_head_with_empty_list_succeeds)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+    
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_add_head(&head, &(simp1.link));
+
+    // assert
+    ASSERT_ARE_EQUAL(void_ptr, &(simp1.link), result);
+    ASSERT_ARE_EQUAL(void_ptr, head.next, result);
+}
+
+/* Tests_SRS_SLIST_07_010: [ `slist_add_head` shall insert `list_entry` at head on success and return an entry to the added entry. ]*/
+TEST_FUNCTION(slist_add_head_with_multiple_entries_in_the_list_succeeds)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+    static simpleItem simp2 = { 2, { NULL } };
+    static simpleItem simp3 = { 3, { NULL } };
+    static simpleItem simp4 = { 4, { NULL } };
+
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+    slist_add(&head, &(simp1.link));
+    slist_add(&head, &(simp2.link));
+    slist_add(&head, &(simp3.link));
+
+    // act
+    PSINGLYLINKEDLIST_ENTRY result = slist_add_head(&head, &(simp4.link));
```

Comment:
>PSINGLYLINKEDLIST_ENTRY result = slist_add [](http://example.com/codeflow?start=4&length=42)

I think this API should be changed to return success/fail

Reply:
done